File: populate-service/src/model/Exame.java
package model;

import java.time.LocalDateTime;
import java.util.Map;
import service.IGenericData;

public class Exame implements IGenericData {

  private int id;
  private String idPaciente;
  private int idSchema;
  private LocalDateTime dataRealizacao;
  private Map<String, Object> dadosPreenchidos; // Representação de JSON como Map
  private String responsavel;
  private String observacoes;

  private Paciente paciente;
  private ExameSchema schema;

  public Exame(int id, String idPaciente, int idSchema, LocalDateTime dataRealizacao,
      Map<String, Object> dadosPreenchidos, String responsavel, String observacoes,
      Paciente paciente, ExameSchema schema) {
    this.id = id;
    this.idPaciente = idPaciente;
    this.idSchema = idSchema;
    this.dataRealizacao = dataRealizacao;
    this.dadosPreenchidos = dadosPreenchidos;
    this.responsavel = responsavel;
    this.observacoes = observacoes;
    this.paciente = paciente;
    this.schema = schema;
  }

  // Getters
  public int getId() {
    return id;
  }

  public String getIdPaciente() {
    return idPaciente;
  }

  public int getIdSchema() {
    return idSchema;
  }

  public LocalDateTime getDataRealizacao() {
    return dataRealizacao;
  }

  public Map<String, Object> getDadosPreenchidos() {
    return dadosPreenchidos;
  }

  public String getResponsavel() {
    return responsavel;
  }

  public String getObservacoes() {
    return observacoes;
  }

  public Paciente getPaciente() {
    return paciente;
  }

  public ExameSchema getSchema() {
    return schema;
  }

  // Setters
  public void setIdPaciente(String idPaciente) {
    this.idPaciente = idPaciente;
  }

  public void setIdSchema(int idSchema) {
    this.idSchema = idSchema;
  }

  public void setDataRealizacao(LocalDateTime dataRealizacao) {
    this.dataRealizacao = dataRealizacao;
  }

  public void setDadosPreenchidos(Map<String, Object> dadosPreenchidos) {
    this.dadosPreenchidos = dadosPreenchidos;
  }

  public void setResponsavel(String responsavel) {
    this.responsavel = responsavel;
  }

  public void setObservacoes(String observacoes) {
    this.observacoes = observacoes;
  }

  public void setPaciente(Paciente paciente) {
    this.paciente = paciente;
  }

  public void setSchema(ExameSchema schema) {
    this.schema = schema;
  }
}

File: webapp/main/php/model/Exame.php
<?php
require_once __DIR__ . '/../dao/IGenericMainData.php';
class Exame implements IGenericMainData
{
    private int $id;
    private string $idPaciente;
    private int $idSchema;
    private DateTime $dataRealizacao;
    private array $dadosPreenchidos; // JSON no banco, array no PHP
    private string $responsavel;
    private string $observacoes;

    private Paciente $paciente;
    private ExameSchema $schema;

    public function __construct(
        int $id,
        string $idPaciente,
        int $idSchema,
        DateTime $dataRealizacao,
        array $dadosPreenchidos,
        string $responsavel,
        string $observacoes,
        Paciente $paciente,
        ExameSchema $schema
    ) {
        $this->id = $id;
        $this->idPaciente = $idPaciente;
        $this->idSchema = $idSchema;
        $this->dataRealizacao = $dataRealizacao;
        $this->dadosPreenchidos = $dadosPreenchidos;
        $this->responsavel = $responsavel;
        $this->observacoes = $observacoes;
        $this->paciente = $paciente;
        $this->schema = $schema;
    }

    // Getters
    public function getId(): int
    {
        return $this->id;
    }
    public function getIdPaciente(): string
    {
        return $this->idPaciente;
    }
    public function getIdSchema(): int
    {
        return $this->idSchema;
    }
    public function getDataRealizacao(): DateTime
    {
        return $this->dataRealizacao;
    }
    public function getDadosPreenchidos(): array
    {
        return $this->dadosPreenchidos;
    }
    public function getResponsavel(): string
    {
        return $this->responsavel;
    }
    public function getObservacoes(): string
    {
        return $this->observacoes;
    }
    public function getPaciente(): Paciente
    {
        return $this->paciente;
    }
    public function getSchema(): ExameSchema
    {
        return $this->schema;
    }

    // Setters
    public function setIdPaciente(string $idPaciente): void
    {
        $this->idPaciente = $idPaciente;
    }
    public function setIdSchema(int $idSchema): void
    {
        $this->idSchema = $idSchema;
    }
    public function setDataRealizacao(DateTime $dataRealizacao): void
    {
        $this->dataRealizacao = $dataRealizacao;
    }
    public function setDadosPreenchidos(array $dadosPreenchidos): void
    {
        $this->dadosPreenchidos = $dadosPreenchidos;
    }
    public function setResponsavel(string $responsavel): void
    {
        $this->responsavel = $responsavel;
    }
    public function setObservacoes(string $observacoes): void
    {
        $this->observacoes = $observacoes;
    }
    public function setPaciente(Paciente $paciente): void
    {
        $this->paciente = $paciente;
    }
    public function setSchema(ExameSchema $schema): void
    {
        $this->schema = $schema;
    }
}
